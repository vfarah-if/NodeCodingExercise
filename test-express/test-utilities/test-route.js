import app from "../server/app";
import supertest from "supertest";
const request = supertest(app);

/*
** REMARKS: Help remove duplication in tests and this wraps the 
**          get without needing to include all the imports
*/
export async function getTest(url) {
    const actualResponse = await request.get(url);
    return actualResponse;
}