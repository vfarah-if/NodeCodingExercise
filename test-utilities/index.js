export const cloneCourse = (course) => {
    const stats = { ...course.stats };
    const result = { ...course, stats };
    return result;
  }