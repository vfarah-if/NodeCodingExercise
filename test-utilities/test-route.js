import app from "../src/app";
import supertest from "supertest";
const request = supertest(app);

/*
** REMARKS: Help remove duplication in tests and this wraps the 
**          get without needing to include all the imports
*/
export async function getRequestTest(url) {
    const actualResponse = await request.get(url);
    return actualResponse;
}