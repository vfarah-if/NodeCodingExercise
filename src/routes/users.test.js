import { getRequestTest } from "../../test-utilities/test-route";

describe("users api", () => {
  it("should get the users json response", async (done) => {
    const actualResponse = await getRequestTest("/users");
    expect(actualResponse).toBeTruthy();
    expect(actualResponse.status).toBe(200);
    expect(actualResponse.text).toMatchSnapshot();
    done();
  });
});
