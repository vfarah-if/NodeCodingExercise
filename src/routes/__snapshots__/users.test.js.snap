// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`users api should get the users json response 1`] = `"[{\\"user\\":{\\"id\\":1,\\"name\\":\\"Jane\\",\\"lastname\\":\\"Doe\\"}},{\\"user\\":{\\"id\\":2,\\"name\\":\\"Jon\\",\\"lastname\\":\\"Doe\\"}},{\\"user\\":{\\"id\\":3,\\"name\\":\\"Michael\\",\\"lastname\\":\\"Caine\\"}},{\\"user\\":{\\"id\\":4,\\"name\\":\\"James\\",\\"lastname\\":\\"Bond\\"}},{\\"user\\":{\\"id\\":5,\\"name\\":\\"Robert\\",\\"lastname\\":\\"Plant\\"}},{\\"user\\":{\\"id\\":6,\\"name\\":\\"Jordan\\",\\"lastname\\":\\"Peterson\\"}}]"`;
